### PR TITLE
[Train] Llama-2 workspace template enabled gce tests with cu121

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/templates/testing/compute_configs/04_finetuning_llms_with_deepspeed/gce_7b.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+head_node_type:
+  name: head_node_type
+  instance_type: n2-standard-32
+
+worker_node_types:
+- name: gpu_worker
+  instance_type: g2-standard-8-nvidia-l4-1
+  min_workers: 0
+  max_workers: 16
+  use_spot: false
```
Comment (aslonnie) [inline doc/source/templates/testing/compute_configs/04_finetuning_llms_with_deepspeed/gce_7b.yaml:13]: nit: add an end-of-line at the end-of-file?

**Review by aslonnie — COMMENTED** [review]
(will let @can-anyscale to review)

**Comment by kouroshHakha** [comment]
rerunning buildkite: https://buildkite.com/ray-project/release-tests-pr/builds/50225

**Comment by xwjiang2010** [comment]
Release test running at: https://buildkite.com/ray-project/release-tests-branch/builds/2085#018a3e17-c9c7-49d8-8da7-c3ca00819dba

**Comment by xwjiang2010** [comment]
gce passing: https://buildkite.com/ray-project/release-tests-pr/builds/52088#018a66a7-b881-4a10-b7ed-6eff9e344c0a

**Review by justinvyu — APPROVED** [review]

File: release/release_tests.yaml [hunk 2]
```diff
@@ -1037,21 +1037,39 @@
   team: ml
   cluster:
     byod:
-      type: gpu
+      type: cu121
+      pip:
+        - torch==2.0.0
```
Comment (can-anyscale) [inline release/release_tests.yaml:1042]: Overal looking good, but can this part be a part of the requirements_byod file?

**Review by aslonnie — COMMENTED** [review]
leaving this for @can-anyscale to review.

File: doc/source/templates/testing/compute_configs/04_finetuning_llms_with_deepspeed/gce_7b.yaml [hunk 3]
```diff
@@ -5,17 +5,11 @@ allowed_azs:
...
 - name: gpu_worker
   instance_type: g2-standard-16-nvidia-l4-1
-  min_workers: 15
-  max_workers: 15
+  min_workers: 0
+  max_workers: 16
   use_spot: false
-  resources:
-    custom_resources: 
-      medium_cpu_mem: 1
```
Comment (can-anyscale) [inline doc/source/templates/testing/compute_configs/04_finetuning_llms_with_deepspeed/gce_7b.yaml:21]: adding new line

File: release/release_tests.yaml (hunk 2 above)
Comment (can-anyscale) [reply release/release_tests.yaml:1042]: can we add a comment that this needs to be in sync with llm-forge

**Review by can-anyscale — APPROVED** [review]

**Pull request merged**